docs/cloud-provider-specifics.md

```diff
@@ -1,5 +1,7 @@
 # Cloud Provider Specifics
 
+> NOTE: Documentation may change between releases. See the [Changelog][20] for links to previous versions of this repository and its docs.
```

Comment:
Ok, redundant is fine. I think we should have the same blob of text in both places (docs may change, see changelog, check out the appropriate tag, etc). WDYT?